web/grunt/karma-config.js

```diff
@@ -0,0 +1,41 @@
+module.exports = function (grunt) {
+    'use strict';
+
+    const unitTests =  grunt.config('config.tests');
+
+    const karmaTasksConfig = {
+        browsers: ['Firefox'],
+        frameworks: ['jasmine', 'browserify'],
+        preprocessors: {
+            [unitTests]: ['browserify'],
+            '<%= config.compiledScriptsDir %>/**/*.js': ['browserify']
+        },
+        browserify: {
+            debug: true,
+            paths: '<%= config.scriptsDir %>'
+        }
+    };
+
+    const configure = function () {
+        grunt.config.merge({
+            karma: {
+                options: {
+                    files: ['<%= config.compiledScriptsDir %>/**/*.js', unitTests]
+                },
+                dev: karmaTasksConfig,
+                ci: Object.assign(
+                    {
+                        options: {
+                            singleRun: true
+                        }
+                    },
+                    karmaTasksConfig
+                )
+            }
+        });
+    };
+
+    return {
+        configure
```

Comment:
That's how grunt plugins are written.

When we need to build complex multi-running things, where we want to require a load of modules then run certain steps of them in sequence, like in some big projects with multiple frontend-ops parts, then this pattern works well. Otherwise, I think it's overcomplicating a simple grunt plugin.